kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -285,7 +295,7 @@ RawBlock DBAdapter::getRawBlock(const BlockId &blockId) const {
 
   // loookup keys |NonProvable|blockid|
   // add to the keyValues
-  const auto &non_provable_key = DBKeyManipulator::genNonProvableBlockDbKey(blockId, Sliver{});
+  const auto &non_provable_key = DBKeyManipulator::genNonProvableDbKey(blockId, Sliver{});
   auto iter = db_->getIteratorGuard();
 
   for (auto kvp = iter->seekAtLeast(non_provable_key);
```

Comment:
Sorry I haven't noticed that before, but it seems this will loop until it encounters a different key type/subtype. I think it should stop also when a new version for the same key type/subtype is encountered.

Could we maybe use `nonProvableKeysForBlock()` for that instead of a raw loop here?

Please correct me if I miss something.

Reply:
Good catch, fixed the condition!
The method `nonProvableKeysForBlock()` returns the `KeyVector` which means we'll need to lookup value for each key.

Reply:
Oh, yes, got it.